e2e/sites.test.ts

```diff
@@ -1,100 +1,54 @@
 import path from 'path';
 import { test, expect } from '@playwright/test';
 import { pathExists } from '../src/lib/fs-utils';
-import { launchApp } from './e2e-helpers';
+import { E2ESession } from './e2e-helpers';
 import MainSidebar from './page-objects/main-sidebar';
 import Onboarding from './page-objects/onboarding';
 import SiteContent from './page-objects/site-content';
-import type { ElectronApplication, Page } from 'playwright';
 
 const skipTestOnWindows = process.platform === 'win32' ? test.skip : test;
 
 test.describe( 'Servers', () => {
-	let electronApp: ElectronApplication;
-	let mainWindow: Page;
-	const siteName = `test-site-${ randomUUID() }`;
-	const tmpSiteDir = path.join( tmpdir(), siteName );
-	const defaultOnboardingSiteName = 'My WordPress Website';
-	const onboardingTmpSiteDir = path.join(
-		tmpdir(),
-		defaultOnboardingSiteName.replace( /\s/g, '-' )
-	);
+	const session = new E2ESession();
 
-	test.beforeAll( async () => {
-		// Print path in error so we can delete it manually if needed.
-		expect( await pathExists( tmpSiteDir ), `Path ${ tmpSiteDir } exists.` ).toBe( false );
-		await fs.mkdir( tmpSiteDir, { recursive: true } );
-
-		// Temporarily launch the app to complete the onboarding process before
-		// interacting with the primary app UI.
-		expect(
-			await pathExists( onboardingTmpSiteDir ),
-			`Path ${ onboardingTmpSiteDir } exists.`
-		).toBe( false );
-		await fs.mkdir( onboardingTmpSiteDir, { recursive: true } );
-		const [ onboardingAppInstance, onboardingMainWindow ] = await launchApp( {
-			E2E_OPEN_FOLDER_DIALOG: onboardingTmpSiteDir,
-		} );
-		const onboarding = new Onboarding( onboardingMainWindow );
-		const sidebar = new MainSidebar( onboardingMainWindow );
-
-		// Await UI visibility before proceeding.
-		await expect( onboarding.heading.or( sidebar.addSiteButton ) ).toBeVisible();
-
-		// If the onboarding heading is present, there are no existing sites and we
-		// must complete the onboarding process.
-		if ( await onboarding.heading.isVisible() ) {
-			await onboarding.selectLocalPathForTesting();
-			await onboarding.continueButton.click();
+	const siteName = 'E2E-Test-Site';
+	const defaultSiteName = 'My WordPress Website';
 
-			const siteContent = new SiteContent( onboardingMainWindow, defaultOnboardingSiteName );
-			await expect( siteContent.siteNameHeading ).toBeVisible( { timeout: 60_000 } );
-		}
-
-		await onboardingAppInstance.close();
-
-		// Relaunch the app but configured to use tmpSiteDir as the path for the local site.
-		[ electronApp, mainWindow ] = await launchApp( { E2E_OPEN_FOLDER_DIALOG: tmpSiteDir } );
+	test.beforeAll( async () => {
+		await session.launch();
+
+		// Complete onboarding before tests
+		const onboarding = new Onboarding( session.mainWindow );
+		await expect( onboarding.heading ).toBeVisible();
+		// Delay clicking the button to avoid errors due to creating the site too early
+		await session.mainWindow.waitForTimeout( 500 );
```

Comment:
I perceive the new sessions ensures the onboarding UX is always presented. 👍🏻 That said, what exactly are we awaiting? How does the artificial delay address it and/or provide additional benefit in comparison to the previous UI visibility implementation? 

Not opposed to this change, merely would like to understand the rationale.

Reply:
Oh, I'm afraid I included this delay by mistake. It comes from tests I'm performing to address failures when running on Windows. I'll remove it asap.

Reply:
I addressed this in https://github.com/Automattic/studio/pull/270/commits/8269a6e6c1b489a574538934b7a56d510caeedcf.